beacon_node/beacon_chain/src/builder.rs

```diff
@@ -958,6 +960,16 @@ where
             );
         }
 
+        info!(
+            log,
+            "start_attestation_simulator_service initialized";
+        );
+        let beacon_chain = Arc::new(beacon_chain);
+        start_attestation_simulator_service(
+            beacon_chain.task_executor.clone(),
+            beacon_chain.clone(),
+        );
```

Comment:
We typically start services like this over in the `client::builder`. There's access to a lot more components there (not that you really need them) and it also means that these services don't start when we're using just the `BeaconChain` for testing.

I'd suggest starting the service just below this one: https://github.com/sigp/lighthouse/blob/441fc1691b69f9edc4bbdc6665f3efab16265c9b/beacon_node/client/src/builder.rs#L840

That would have the upside that you don't need to return an `Arc<BeaconChain<...>>` here anymore.